src/printer.rs

```diff
@@ -103,15 +112,25 @@ impl Printer {
         }
     }
 
+    fn print_javascript_text(&mut self, text: &str) -> io::Result<()> {
+        if valid_json(text) {
+            self.print_json_text(text, false)
+        } else {
+            self.print_syntax_text(text, "js")
+        }
+    }
```

Comment:
if this is going to be only called from `print_body_text()`, then it is possible to move the `valid_json()` check to when pattern matching the content-type

```rust
 match content_type {
    /* code omitted */
    Some(ContentType::JavaScript) if valid_json(body) => self.print_javascript_text(body),
    Some(ContentType::JavaScript) => self.print_javascript_text(body),
    Some(ContentType::Text) if valid_json(body) => self.print_json_text(body, false),
     _ => self.buffer.print(body),
}
```